cmd/talosctl/cmd/talos/edit.go

```diff
@@ -177,5 +179,6 @@ or 'notepad' for Windows.`,
 func init() {
 	editCmd.Flags().StringVar(&editCmdFlags.namespace, "namespace", "", "resource namespace (default is to use default namespace per resource)")
 	helpers.AddModeFlags(&editCmdFlags.Mode, editCmd)
```

Comment:
I might be wrong, but feels like `--dry-run` for `talosctl edit mc` is a bit weird, as you make changes, save them, and basically changes are lost?  (we can keep it for consistency sake still)

But at the same time `--dry-run` for `talosctl patch mc` might be interesting, as it gives us the ability to see the patch being applied, and it's easy to remove `--dry-run` after that to actually apply the changes